### ISSUE TITLE
catkin make fails in ROS Melodic

### ISSUE THREAD

**Issue by gisulee-mr** [code-iai/pico_flexx_driver#21]
I have a problem on catkin_make is unable to go through. This is the error message

ubuntu 18.04 64bit

![image](https://user-images.githubusercontent.com/37143542/57142475-183d8c00-6dac-11e9-90d0-29c521d49c01.png)

**Comment by mintar** [comment]
It looks like you are using the 64 bit x86 version of libroyale on a odroid. You should use the ARM version instead.

**Comment by Arag24** [comment]
Hi, I'm also working with ROS Melodic // 64bit.
It looks like I'm facing similar thing...

```
damien@ubuntu:~/catkin/pmd$ catkin_make 
Base path: /home/damien/catkin/pmd
Source space: /home/damien/catkin/pmd/src
Build space: /home/damien/catkin/pmd/build
Devel space: /home/damien/catkin/pmd/devel
Install space: /home/damien/catkin/pmd/install
####
#### Running command: "make cmake_check_build_system" in "/home/damien/catkin/pmd/build"
####
####
#### Running command: "make -j4 -l4" in "/home/damien/catkin/pmd/build"
####
[  9%] Automatic MOC for target royale_in_ros
[ 18%] Automatic MOC for target royale_nodelet
[ 18%] Built target royale_in_ros_autogen
[ 18%] Built target royale_nodelet_autogen
Scanning dependencies of target royale_in_ros
Scanning dependencies of target royale_nodelet
[ 27%] Building CXX object libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_nodelet.dir/src/RoyaleInRos.cpp.o
[ 36%] Building CXX object libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_in_ros.dir/src/RoyaleInRos.cpp.o
[ 45%] Building CXX object libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_nodelet.dir/src/RoyaleControl.cpp.o
[ 54%] Building CXX object libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_nodelet.dir/royale_nodelet_autogen/mocs_compilation.cpp.o
[ 63%] Building CXX object libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_in_ros.dir/src/RoyaleControl.cpp.o
[ 72%] Building CXX object libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_in_ros.dir/src/main.cpp.o
[ 81%] Linking CXX shared library /home/damien/catkin/pmd/devel/lib/libroyale_nodelet.so
[ 90%] Building CXX object libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_in_ros.dir/royale_in_ros_autogen/mocs_compilation.cpp.o
/usr/bin/ld:/home/damien/catkin/pmd/src/libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/royale/bin/libroyale.so: file format not recognized; treating as linker script
/usr/bin/ld:/home/damien/catkin/pmd/src/libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/royale/bin/libroyale.so:0: syntax error
collect2: error: ld returned 1 exit status
libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_nodelet.dir/build.make:199: recipe for target '/home/damien/catkin/pmd/devel/lib/libroyale_nodelet.so' failed
make[2]: *** [/home/damien/catkin/pmd/devel/lib/libroyale_nodelet.so] Error 1
CMakeFiles/Makefile2:2814: recipe for target 'libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_nodelet.dir/all' failed
make[1]: *** [libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_nodelet.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Linking CXX executable /home/damien/catkin/pmd/devel/lib/royale_in_ros/royale_in_ros
/usr/bin/ld:/home/damien/catkin/pmd/src/libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/royale/bin/libroyale.so: file format not recognized; treating as linker script
/usr/bin/ld:/home/damien/catkin/pmd/src/libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/royale/bin/libroyale.so:0: syntax error
collect2: error: ld returned 1 exit status
libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_in_ros.dir/build.make:225: recipe for target '/home/damien/catkin/pmd/devel/lib/royale_in_ros/royale_in_ros' failed
make[2]: *** [/home/damien/catkin/pmd/devel/lib/royale_in_ros/royale_in_ros] Error 1
CMakeFiles/Makefile2:793: recipe for target 'libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_in_ros.dir/all' failed
make[1]: *** [libroyale-3.21.1.70-LINUX-x86-64Bit/samples/cpp/sampleROS/CMakeFiles/royale_in_ros.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
Invoking "make -j4 -l4" failed
```

Is there something I did not correctly, too ?

**Comment by Arag24** [comment]
I can close the previous one.

Problem was due to Windows zip file. It was faulty (losing some links) causing the symptom. Indeed, I'm using Virtual Machine and I was sharing a Windows "unzipped foilder to my Ubuntu VM).
It has been solved sharing Zip to linux VM and unzipping directly inside Ubuntu. 

Version installed 3.20.0.62 [should also work 3.21. but I didn't have zip file currently on this revision today].

However I need a bit of help anyway. If I can see my PMD Monstar running, I didn't succeed to see any Point Cloud. Following the README, I selected `pico_flex_link` for any Frame, however I don't know if this is publish as I got error status every time. I'm definitely not a very familiar with ROS, any help is welcome.
![image](https://user-images.githubusercontent.com/27350352/57197623-df4b1600-6f69-11e9-9b50-c7b963f40a15.png)
